### DISCUSSION TITLE
Работа с модалками

### DISCUSSION THREAD

**Discussion by svyatoslavpavlov** [VKCOM/VKUI#7671] [Q&A] [answered]
Привет! Мне очень непривычна работа с модалками в VKUI
Допустим, я хочу сделать компонент-селектор, который выглядит как кнопка и открывает модальное окно. В случае с vkui я не могу инкапсулировать логику внутри компонента, я должен поменять еще внешний слой. С точки зрения архитектуры выглядит как-то неправильно.

Возможно, все дело в том, что я мобильный разработчик и текущая ситуация кажется мне совсем странной, но взгляните, как можно работать с модалками в jetpack compose или swiftui, там можно использовать их и как часть навигации, и как часть компонента. 

Возможно, вы также мне подскажете, как грамотно будет в текущей реализации возвращать значение из модалки обратно в селектор?

**Comment by inomdzhon** [comment]
Привет,

В задаче:

- #7529

идём к тому, чтобы исправить эту проблему

Начиная с **v7.0.0** можно будет рендерить модалку из любого места

**Comment by inomdzhon** [comment, marked as answer]
Поправили проблему в **v7**. Теперь можно инкапсулировать логику через прямое использование `ModalPage` или `ModalCard`.

Последняя версию на текущий момент [v7.0.1](https://github.com/VKCOM/VKUI/releases/tag/v7.0.1).
